client/v3/v3.go

```diff
@@ -18,16 +18,25 @@ type Client struct {
 
 // NewV3Client return a client to operate V3 resources
 func NewV3Client(credentials client.Credentials) (*Client, error) {
-	c, err := client.NewClient(&credentials, userAgent, absolutePath, false)
-
-	if err != nil {
-		return nil, err
+	var baseClient *client.Client
+
+	// check if all required fields are present. Else create an empty client
+	if credentials.Username != "" && credentials.Password != "" && credentials.Endpoint != "" {
+		c, err := client.NewClient(&credentials, userAgent, absolutePath, false)
+		if err != nil {
+			return nil, err
+		}
+		baseClient = c
+	} else {
+		baseClient = &client.Client{UserAgent: userAgent, Error: "nutanix pc Client is missing. " +
```

Comment:
```suggestion
		baseClient = &client.Client{UserAgent: userAgent, Error: "nutanix PC (Prism Central) client is missing. " +
```

Reply:
Pass these required parameters from ConfigureProvider function. Otherwise in future we need to make changes in this package as well if we change auth schema for any client. 

Make changes in every client.

Even we can think of creating auth fields as parameters of credentials. We can create error msg dynamically once we know the keys or fields of auth schema.

Make changes for all clients.